From 6132342e02c6d0aaf6a0385a37f0e2b2468513d3 Mon Sep 17 00:00:00 2001
From: Michal Simek <michal.simek@xilinx.com>
Date: Mon, 12 Mar 2018 16:13:57 +0100
Subject: [PATCH 0727/1852] Bluetooth: hci_ll: Extend timeout between start and
 communication

commit 636313c7b98a7773d3a127141f92ac6ed09fa7f4 from
https://github.com/Xilinx/linux-xlnx.git

Extend timeout between bt enable and communication with chip.
Flow control is automatic and driver is responding that device is ready
to accept messages but it doesn't need to be truth. Wait more time to
ensure that communication cat start.

Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/bluetooth/hci_ll.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/bluetooth/hci_ll.c b/drivers/bluetooth/hci_ll.c
index c04f5f9e1ed0..404f0e3e1b76 100644
--- a/drivers/bluetooth/hci_ll.c
+++ b/drivers/bluetooth/hci_ll.c
@@ -606,6 +606,7 @@ static int ll_setup(struct hci_uart *hu)
 		gpiod_set_value_cansleep(lldev->enable_gpio, 0);
 		msleep(5);
 		gpiod_set_value_cansleep(lldev->enable_gpio, 1);
+		msleep(500);
 		err = serdev_device_wait_for_cts(serdev, true, 200);
 		if (err) {
 			bt_dev_err(hu->hdev, "Failed to get CTS");
-- 
2.31.1

